File: support_tools/swigwin-3.0.12/Examples/test-suite/php/li_carrays_cpp_runme.php
<?php
require "tests.php";
require "li_carrays_cpp.php";

// Check functions.
check::functions(array(new_intarray,delete_intarray,intarray_getitem,intarray_setitem,doublearray_getitem,doublearray_setitem,doublearray_cast,doublearray_frompointer,xyarray_getitem,xyarray_setitem,xyarray_cast,xyarray_frompointer,delete_abarray,abarray_getitem,abarray_setitem,shortarray_getitem,shortarray_setitem,shortarray_cast,shortarray_frompointer,sum_array));

// Check classes.
// NB An "li_carrays_cpp" class is created as a mock namespace.
check::classes(array(li_carrays_cpp,doubleArray,AB,XY,XYArray,shortArray));

// Check global variables.
check::globals(array(xy_x,xy_y,globalxyarray,ab_a,ab_b,globalabarray));

$d = new doubleArray(10);

$d->setitem(0, 7);
$d->setitem(5, $d->getitem(0) + 3);
check::equal($d->getitem(0) + $d->getitem(5), 17., "7+10==17");

check::done();
?>

File: support_tools/swigwin-3.0.12/Examples/test-suite/perl5/li_carrays_cpp_runme.pl
#!/usr/bin/perl
use strict;
use warnings;
use Test::More tests => 34;
BEGIN { use_ok('li_carrays_cpp') }
require_ok('li_carrays_cpp');

# array_class
{
  my $length = 5;
  my $xyArray = new li_carrays_cpp::XYArray($length);
  for (my $i=0; $i<$length; $i++) {
    my $xy = $xyArray->getitem($i);
    $xy->{x} = $i*10;
    $xy->{y} = $i*100;
    $xyArray->setitem($i, $xy);
  }
  for (my $i=0; $i<$length; $i++) {
    is($xyArray->getitem($i)->{x}, $i*10);
    is($xyArray->getitem($i)->{y}, $i*100);
  }
}

{
  # global array variable
  my $length = 3;
  my $xyArrayPointer = $li_carrays_cpp::globalXYArray;
  my $xyArray = li_carrays_cpp::XYArray::frompointer($xyArrayPointer);
  for (my $i=0; $i<$length; $i++) {
    my $xy = $xyArray->getitem($i);
    $xy->{x} = $i*10;
    $xy->{y} = $i*100;
    $xyArray->setitem($i, $xy);
  }
  for (my $i=0; $i<$length; $i++) {
    is($xyArray->getitem($i)->{x}, $i*10);
    is($xyArray->getitem($i)->{y}, $i*100);
  }
}

# array_functions
{
  my $length = 5;
  my $abArray = li_carrays_cpp::new_ABArray($length);
  for (my $i=0; $i<$length; $i++) {
    my $ab = li_carrays_cpp::ABArray_getitem($abArray, $i);
    $ab->{a} = $i*10;
    $ab->{b} = $i*100;
	li_carrays_cpp::ABArray_setitem($abArray, $i, $ab);
  }
  for (my $i=0; $i<$length; $i++) {
    is(li_carrays_cpp::ABArray_getitem($abArray, $i)->{a}, $i*10);
    is(li_carrays_cpp::ABArray_getitem($abArray, $i)->{b}, $i*100);
  }
  li_carrays_cpp::delete_ABArray($abArray);
}

{
  # global array variable
  my $length = 3;
  my $abArray = $li_carrays_cpp::globalABArray;
  for (my $i=0; $i<$length; $i++) {
    my $ab = li_carrays_cpp::ABArray_getitem($abArray, $i);
    $ab->{a} = $i*10;
    $ab->{b} = $i*100;
	li_carrays_cpp::ABArray_setitem($abArray, $i, $ab);
  }
  for (my $i=0; $i<$length; $i++) {
    is(li_carrays_cpp::ABArray_getitem($abArray, $i)->{a}, $i*10);
    is(li_carrays_cpp::ABArray_getitem($abArray, $i)->{b}, $i*100);
  }
}
